crypto/err/err.c

```diff
@@ -512,24 +513,32 @@ void ERR_error_string_n(unsigned long e, char *buf, size_t len)
      * directly instead.
      */
     r = ERR_GET_REASON(e);
+#ifndef OPENSSL_NO_ERR
     if (ERR_SYSTEM_ERROR(e)) {
         if (openssl_strerror_r(r, rsbuf, sizeof(rsbuf)))
             rs = rsbuf;
     } else {
         rs = ERR_reason_error_string(e);
     }
+#endif
     if (rs == NULL) {
         BIO_snprintf(rsbuf, sizeof(rsbuf), "reason(%lu)", r);
         rs = rsbuf;
     }
 
-    BIO_snprintf(buf, len, "error:%08lX:%s:%s:%s", e, ls, "", rs);
+    BIO_snprintf(buf, len, "error:%08lX:%s:%s:%s", e, ls, func, rs);
```

Comment:
Open question: ERR_print_error_cb() had dropped the output of `e` (the error code).  If that's still a good idea (which I think it is), then it should be dropped here as well.

Reply:
Why do you think it is a good idea?

Reply:
Hmm, actually what was the reason for dropping that? It happened with #9765 merge but there was not clear reasoning behind that. In some (although rather obscure) cases it might help with diagnostics.

Reply:
I can't see any discussion about that in #9765, which is where that happened.  However, I've long found it useless as soon as we're talking dynamically added "library" numbers (which are generated for providers), because `openssl errstr` most probably will give you utter nonsense for them.

Reply:
Okie, we can certainly leave it like this now, and get back to the question if anyone finds the code disturbing or confusing.  Until then, [shrug]